x/auth/account.go

```diff
@@ -121,12 +139,339 @@ func (acc *BaseAccount) SetSequence(seq uint64) error {
 	return nil
 }
 
-//----------------------------------------
-// Wire
+// SpendableCoins returns the total set of spendable coins. For a base account,
+// this is simply the base coins.
+func (acc *BaseAccount) SpendableCoins(_ time.Time) sdk.Coins {
+	return acc.GetCoins()
+}
+
+// TrackDelegation performs delegation accounting. For a base account it simply
+// returns the base coins minus the desired delegation amount.
+func (acc *BaseAccount) TrackDelegation(blockTime time.Time, amount sdk.Coins) sdk.Coins {
+	return acc.GetCoins().Minus(amount)
+}
+
+// TrackUndelegation performs undelegation accounting. For a base account it
+// simply returns the base coins plus the undelegation amount.
+func (acc *BaseAccount) TrackUndelegation(amount sdk.Coins) sdk.Coins {
+	return acc.GetCoins().Plus(amount)
```

Comment:
Here we don't modify the account (see below comments).

Reply:
Updated.